restapi/src/main/java/io/stargate/web/resources/Db.java

```diff
@@ -36,7 +38,7 @@
   private final DataStore dataStore;
   private final AuthenticationService authenticationService;
   private final AuthorizationService authorizationService;
-  private final LoadingCache<String, AuthenticationSubject> docsTokensToRoles =
+  private final LoadingCache<TokenAndHeaders, AuthenticationSubject> docsTokensToRoles =
```

Comment:
If we use _all_ headers at the key, is it going to be different for every request? I suspect some of the HTTP headers may be different in each request... or at least likely to be different, e.g. `Content-Length`... WDYT?

Reply:
Again, if we store auth data inside `AuthenticationSubject` we should be able to continue using only the token as the key here, I believe.

Reply:
Good point, we should construct key based only on token and tenant_id and Host headers (if they are present)